### ISSUE TITLE
createHistory does not work to set basename

### ISSUE THREAD

**Issue by pkosenko** [ReactTraining/react-router#3857]
I attempted to create a basename to allow me to run my React application in a subfolder.  However, despite correct import of createHistory, I keep getting the aggravating error "TypeError: createHistory is not a function"

import { useRouterHistory } from 'react-router'
import { createHistory } from 'history'

const history = useRouterHistory(createHistory)({
  basename: '/base-path'
});

Then in the JSX render:

ReactDOM.render(
  &lt;Provider store={librivoxStore}&gt;
    &lt;Router history={history}&gt;
      &lt;Route path="/" component={App2}/&gt;
    &lt;/Router &gt;
  &lt;/Provider&gt;, document.getElementById("appdiv"));

## Version
2.0.0

## Test Case
http://jsbin.com/sacerobuxi/edit?html,js,output

**Comment by timdorr** [comment]
You need to install history 2.x into your project.

**Comment by pkosenko** [comment]
I do have history 2.x installed.  But do you have a better idea of where the breaking point is?  Do I need an EARLIER version than 2.7.4?  It isn't a matter of createHistory being undefined.  It is there in the project, but the console log is telling me that it is "not a function".

$ npm -v history
2.7.4

**Comment by pkosenko** [comment]
Wow . . . I looked into my node_modules/history folder, which apparently has the lastest code, and it seems, indeed, to be missing the createHistory.js file.  The latest version of history seems to have been slimmed down a lot.  createHistory.js is in history 2.0.0.  So I will try that and see if it resolves my problem.  But it would have been nice if Webpack had done a version check for me (although I don't know if there is a versioning plugin that would do something like that).

**Comment by pkosenko** [comment]
NO LUCK.  I am still getting the "TypeError: createHistory is not a function" error with history 2.0.0 installed rather than 2.7.4.  There is something ELSE broken in REACT ROUTER that it seems no longer likes using createHistory().  And I am guessing that as a result, the documented example of how to 
When I look into the bundle.js file produced by Webpack, I see a lot of commentary about things being deprecated.

bundle.js:

	function useBasename(createHistory) {
	  return function () {
	    var options = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];
	    var history = createHistory(options);     <--------------  CONSOLE.LOG POINT HERE AS ERROR
	    var basename = options.basename;
	    var checkedBaseHref = false;
	    function checkBaseHref() {
	      if (checkedBaseHref) {
	        return;
	      }

**Comment by timdorr** [comment]
That's not the command to show the version of a module. That's showing you npm's version. Run `npm ls history` and see what versions are installed in your node_module tree.

**Comment by timdorr** [comment]
Also, the correct version of history to install is 2.1.2. You should have that and react-router@2.8.1 (released the other day).

**Comment by pkosenko** [comment]
Thanks.  I have solved the problem.  It was dependency hell. 

When I checked module versions and dependencies, I noticed that I had a copy of "redux-react-router" that was supplying its OWN dated version of history@1.9.0.

I was THINKING of using redux-react-router but HAVEN'T YET.  So I uninstalled it and the "TypeError: createHistory is not a function" error disappeared.

**The UPSHOT is . . . beware of installing "redux-react-router" with the latest react-router, since it is supplying a dated version of history.  Someone should cross-post this to "redux-react-router" so that they UPDATE their module to make it conform to the latest "react-router" module.**

This is how I actually found the history module conflict:
$ npm ls > package_versions.txt  
npm ERR! extraneous: history@1.9.0 c:\Users\pkosenko\sites\devdesktop\jstest\react_librivox_search
\node_modules\redux-react-router\node_modules\history

A quick check of dependencies showed:

Top level of node_modules has a "history" of the correct 2.1.2 version that you note (I installed it explicitly):

{
  "name": "history",
  "version": "2.1.2",
  .....
}

And a proper react-router:

{
  "name": "react-router",
  "version": "2.8.1",
  .....
}

And react-router also has its OWN version of history, CORRECT AT 2.1.2
So . . .  I am not sure that a top-level history module is even needed, although it causes no problems once the redux-react-router  history@1.9.0 is removed.

In react-router:

{
  "name": "history",
  "version": "2.1.2",
  ....
}
